teselagen/api/build_client.py

```diff
@@ -0,0 +1,342 @@
+                url=url,
+                headers=self.headers,
+            )
+            assert response['content'] is not None  # noqa: S101
+            output_aliquot = cast(AliquotRecord, json.loads(response['content']))
+
+        except Exception as _exc:  # noqa: F841
+            # fallback to bruteforce if an error occurs
+            output_aliquot = self._get_record(
+                get_records=self.get_aliquots,
+                record_id=aliquot_id,
+            )
+
+        return output_aliquot
+
+    def get_aliquots(
+            self,
+            pageNumber: str | int = '1',  # noqa: N803
+            pageSize: str | int = DEFAULT_PAGE_SIZE,
+            sort: str = '-updatedAt',
+            gqlFilter: str = '',  # TODO(diegovalenzuelaiturra): Chech default parameter (if any).
+    ) -> List[AliquotRecord]:
+        """This is a paged entrypoint for returning many aliquot records.
+
+        Args:
+            pageNumber (str): 1 based paging parameter. Default: `"1"`.
+
+            pageSize (str): size of each page returned. Default: `"100"`.
+
+            sort (str): field to sort on, default is id. Default: `"-updatedAt"`.
+
+            gqlFilter (str): A `graphql` filter to apply to the data. Example:
+
+        ```GraphQL
+                { "sample.material.name" : ["PCR53.1", "Sequence2"] } or { "id": ["1", "10", "22"] }
+        ```
+
+        Returns:
+            List[AliquotRecord]: List of aliquot records.
+        """
+        params: GetAliquotsQueryParams = {
+            'pageNumber': str(pageNumber),
+            'pageSize': str(pageSize),
+            'sort': str(sort),
+            'gqlFilter': str(gqlFilter),
+        }
+
+        response = get(
+            url=self.aliquots_url,
+            headers=self.headers,
+            params=params,
+        )
+
+        assert response['content'] is not None
+
+        return cast(List[AliquotRecord], json.loads(response['content']))
+
+    def get_sample(
+        self,
+        sample_id: str | int,
+    ) -> SampleRecord:
+        """This function returns a single sample by id.
+
+        Args:
+            sample_id (str): The id of the sample to return.
+
+        Returns:
+            SampleRecord: Sample record.
+        """
+        output_sample: SampleRecord = None  # {}
+
+        try:
+            url: str = self.sample_url.format(str(sample_id))
+            response: ResponseDict = get(
+                url=url,
+                headers=self.headers,
+            )
+            assert response['content'] is not None  # noqa: S101
+            output_sample = cast(SampleRecord, json.loads(response['content']))
+
+        except Exception as _exc:  # noqa: F841
+            # fallback to bruteforce if an error occurs
+            output_sample = self._get_record(
+                get_records=self.get_samples,
+                record_id=sample_id,
+            )
+
+        return output_sample
+
+    def get_samples(
```

Comment:
It can be very useful to add some simple fields to make filtering easier. For example, adding `name` as a parameter to filter the query. This can be combined with the gqlFilter parameter. I mention this (filtering by name) because it was mention explicitly by the client (https://github.com/TeselaGen/dtu-bioengineering-support/issues/10). An example of how this can be implemented is at the `get_designs` method in `design_client.py`

Reply:
@juanneilson Are the names guaranteed to be unique?

Reply:
I think it is not necessary as the filter should reduce the number of query outputs and there's no need to return a single register.

Reply:
First, that may be out of the scope of this ticket.

Second, I think we should not add additional complexity to the methods that serve as interfaces to the API, and their behaviors should not differ.

Also, if the user can already do what they require with existing endpoints, maybe we should provide/request better documentation or examples where appropriate.

Reply:
Let's discuss this for future developments